### ISSUE TITLE
DeepL всё ещё в не рабочем состоянии

### ISSUE THREAD

**Issue by RedCapcha** [OneMoreGres/ScreenTranslator#164]
Заметил, что на перевод DeepL вышло обновление и решил снова попользовать любимую прогу. В ответ получил следующее:
https://static.deepl.com/js/translator_late.min.$6e9607.js: 44 DeepL Translator 
https://static.deepl.com/js/translator_late.min.$6e9607.js: 44 Welcome to DeepL.com!
https://static.deepl.com/js/translator_late.min.$6e9607.js: 44 Uncaught (in promise) TypeError: Cannot read property 'getVoices' of undefined
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Module already defined 'LMT_UI_SystemNotification'
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Module already defined 'FeatureManager'
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [FeatureManager.defineType] /translator/core/selectLangUI is already defined!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="284", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="270", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="25", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="1", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 0:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 3376ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="261", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 1:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 10962ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [FeatureManager] Required features could not be resolved within 5000ms 
Required: /translator/core/selectedLanguages,/translator/core/textareas,/translator/core/requestHandling 
Missing:  
Required by: "/translator/common/speakTargetText"
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [FeatureManager] Required features could not be resolved within 5000ms 
Required: /translator/webTranslatorCore,/translator/core/config,/translator/sourceToolbar/toolbar,/translator/environment/localizedTexts 
Missing: /translator/sourceToolbar/toolbar 
Required by: "/translator/sourceToolbar/undoRedoButtons"
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [FeatureManager] Required features could not be resolved within 5000ms 
Required: /translator/core/config,/translator/core/domElements,/translator/environment/localizedTexts,/translator/webTranslatorCore,/translator/common/speakTargetText 
Missing: /translator/common/speakTargetText 
Required by: "/translator/targetToolbar/toolbar"
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 2:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 15451ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 3:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [U.remove] Callback not found function(){return u.waitForPendingRequests().then(function(){var t=Date.now()-a;setTimeout(function(){var e,n,r;v&&v.isRequestPending()?null===(r=v)||void 0===r||r.isRequestPending.onValueChanged.push(function(){var e,n;return C(l(l({},O),{},{dictionaryTriggered:null!==(e=null===(n=v)||void 0===n?void 0:n.isDictionaryAvailable())&&void 0!==e&&e,completeTranslationDuration:t},f.getTrackingData(O.textSessionId))),"REMOVE"}):C(l(l({},O),{},{dictionaryTriggered:null!==(e=null===(n=v)||void 0===n?void 0:n.isDictionaryAvailable())&&void 0!==e&&e,completeTranslationDuration:t},f.getTrackingData(O.textSessionId)))})}),"REMOVE"}
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="244", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="345", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="215", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 Sending to DAP went wrong (response-code="4", event-id="50", page-id="110")
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://www.deepl.com/translator: 0 Autofocus processing was blocked because a document already has a focused element.
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 0:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 4191ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 1:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 10652ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 0:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 3991ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 1:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 10631ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 2:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 13722ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 2:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Trying again in 13855ms!
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 3:  [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [object Object]
https://static.deepl.com/js/sentry.min.$de3bf5.js: 1 [sentenceSplitting] Error during splitting request (retryNr 3:  [object Object]

**Comment by RedCapcha** [comment]
И хотелось узнать чем сейчас так занят автор, новый проект?

**Comment by OneMoreGres** [comment]
Попробуйте включить загрузку картинок в окне переводчика до первого перевода.

По поводу того, чем занимаюсь. Как у всех: работа, личная жизнь. Просто приоритеты меняются. И тут в ридми и на сайте в описании переводчика написано, что проект практически заброшен. Код открыт, так что продолжить может кто угодно

**Comment by RedCapcha** [comment]
> Попробуйте включить загрузку картинок в окне переводчика до первого перевода.
> 
> По поводу того, чем занимаюсь. Как у всех: работа, личная жизнь. Просто приоритеты меняются. И тут в ридми и на сайте в описании переводчика написано, что проект практически заброшен. Код открыт, так что продолжить может кто угодно

Могу пожелать удачи и в работе и в личной жизни, но ты прямо-таки подводишь международное сообщество любителей игр для взрослых.

**Comment by OneMoreGres** [comment]
> Могу пожелать удачи и в работе и в личной жизни, но ты прямо-таки подводишь международное сообщество любителей игр для взрослых.

:DD Спасибо)

С DeepL одними скриптами перевода проблему решить не получается, а с программой дольше разбираться. Давно не обновлял - все протухло уже..


> И к сожалению я так и не понял где включить эту загрузку картинок, пункт Показывать изображение?

В окне переводчика вторая кнопка (иконка) справа (`Загружать изображения`).

Извиняюсь, что долго не отвечаю. Почту редко проверяю

**Comment by RafeBordo** [comment]
Странно он себя ведет иногда, вчера было все ок, сегодня при попытке перевести, вот такая беда (поле с переводом пустым остается), потому и не выдает перевод.
![image](https://github.com/OneMoreGres/ScreenTranslator/assets/68481959/ef57c686-da01-481c-a4a9-e6615b04680e)

И это текстом, если надо
userscript:deepl.js: 28 start translate deep en ru
userscript:deepl.js: 78 setting url https://www.deepl.com/translator#en/ru/deep
https://static.deepl.com/gatsby/f1ca995fe4bac1ec2cc35319d8b08d0c249b1d30-4a3f69d695cfd32a2950.js: 1 Uncaught (in promise) TypeError: Cannot read property 'getVoices' of undefined
https://static.deepl.com/gatsby/f1ca995fe4bac1ec2cc35319d8b08d0c249b1d30-4a3f69d695cfd32a2950.js: 1 Uncaught (in promise) TypeError: Cannot read property 'getVoices' of undefined
https://static.deepl.com/gatsby/c8f7fe3b0e41be846d5687592cf2018ff6e22687-74a8a9c43bf5d249ed11.js: 14 [U.remove] Callback not found ()=>(l.waitForPendingRequests().then(()=>{let r=Date.now()-o;setTimeout(()=>{var n,s,o;g&&g.isRequestPending()?null===(o=g)||void 0===o||o.isRequestPending.onValueChanged.push(()=>{var n,s;return t({...e,dictionaryTriggered:null!==(n=null===(s=g)||void 0===s?void 0:s.isDictionaryAvailable())&&void 0!==n&&n,completeTranslationDuration:r,...d.getTrackingData(e.textSessionId)}),"REMOVE"}):t({...e,dictionaryTriggered:null!==(n=null===(s=g)||void 0===s?void 0:s.isDictionaryAvailable())&&void 0!==n&&n,completeTranslationDuration:r,...d.getTrackingData(e.textSessionId)})})}),"REMOVE")
: 0 The resource https://static.deepl.com/gatsby/page-data/en/translator/page-data.2cb63b71.json was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

**Comment by OneMoreGres** [comment]
Да, даже вручную и с картинками не хочет :( Надо попробовать собрать с новой версией встроенного браузера, но я хз когда получится это сделать =/

**Comment by RedCapcha** [comment]
В теме автопереводчика на юнити нашел раздел относящийся к DeepL. Там обсуждали, что у сервиса изменился код и для того чтобы интеграция работала нужно несколько каких то параметров изменить. Я сам в этом не тунь тунь, так что передаю всё в руки профессионала. Вот этот раздел https://github.com/bbepis/XUnity.AutoTranslator/issues/431

**Comment by OneMoreGres** [comment]
Посмотрел, спасибо.
Там подход в целом несколько другой, так что напрямую использовать не получится. Если только целиком подход скопировать. Но это долго и хз, что в итоге получится...
Еще можно не через сайт переводить, а напрямую через сервис, получив у них ключ. Правда бесплатный быстро закончится, а платный по нашим картам не купишь и деньги будет съедать, да и файлы со скриптами придется вручную править. Но по крайней мере сделать проще. Готовы заморочиться? :)

**Comment by RedCapcha** [comment]
Я проверял и в России нельзя получить их ключ, там толи номер иностранный нужен, толи ещё что-то. Я всё ещё надеюсь что вы возьмётесь за решение проблемы.

**Comment by OneMoreGres** [comment]
Да, я тоже на этом этапе застопорился. Прототип скрипта есть, но проверить его не могу.
По другим вариантам пока хз, как быстро сделать
